src/validation.cpp

```diff
@@ -316,6 +316,7 @@ bool CheckSequenceLocks(const CTransaction &tx, int flags, LockPoints* lp, bool
                 }
             }
             lp->maxInputBlock = tip->GetAncestor(maxInputHeight);
+            assert(lp->maxInputBlock != nullptr);
```

Comment:
Just `assert(lp->maxInputBlock);`?

Reply:
@promag While legal, I believe for security-critical code implicit conversions such as these should be discouraged. See also http://releases.llvm.org/3.8.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-implicit-bool-cast.html

Reply:
I see why implicit conversions should be discouraged for floating point numbers and integers or bools, but I don't see any danger or downsides of `assert(some_pointer);`